ntoskrnl/mm/ARM3/pool.c

```diff
@@ -1658,8 +1661,8 @@ MmFreeMappingAddress(
     }
 
     /* We must have a size */
-    SizeInPages = PointerPte[0].u.List.NextEntry;
-    if (SizeInPages < 3)
+    SizeInPages = PointerPte[0].u.Long >> 1;
+    if (PointerPte[0].u.Long < (3 << 1))
```

Comment:
Please revert

Reply:
Reverted. Now we have 47 failed tests.

![w2k3sp2_kmtest_MmReservedMapping_fails](https://github.com/user-attachments/assets/922407fa-007b-413e-b596-8ac19059275c)

Can you please help me with fixing these new failing tests?

Reply:
Fixed with the latest revision to the code.